starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,38 @@
-function QueueDataStructure () {
-}
+class QueueDataStructure {
+    constructor() {
+        this.queueControl = []
+        this.MAX_SIZE = 8
+    }
+
+    isEmpty() {
+        if (this.queueControl.length == 0) {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    canEnqueue() {
+        if (this.queueControl.length === this.MAX_SIZE) {
+            return false
+        }
+        return true
+    }
+
+    enqueue(elm) {
+        if (this.canEnqueue()) {
+            this.queueControl.unshift(elm)
+            return this.queueControl
+        } else {
+            return "Queue Overflow"
+        }
+    }
+
+    dequeue() {
+        if (this.queueControl.length > 0) {
```

Comment:
Esta linea está bien, pero compruebas si el array esta vacio o no, que eso lo haces en la funcion isEmpty().
Sería mejor llamar a la función que te lo comprueba y asi no ponerlo todo el rato.
Quería algo asi:
if (!this.isEmpty()) {

Reply:
Ostias, con el cansancio y hacer el pull request rápido para irme a dormir se me pasó poner el MAD DE y mi nombre. Lo acabo de modificar jajaja ¿me cuenta como entregado no? Que cabeza la mía